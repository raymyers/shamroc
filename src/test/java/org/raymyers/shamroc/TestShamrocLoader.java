package org.raymyers.shamroc;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.io.InputStream;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class TestShamrocLoader {
    @Test
    void basicParse() throws IOException {
        InputStream stream = this.getClass().getResourceAsStream("Shamroc.shamroc");
        ShamrocLoader subject = new ShamrocLoader(stream);
//        System.out.println(subject.getSyntaxErrors());
        assertTrue(subject.valid());
        Approvals.verify(subject.getStringTree());
    }

    @Test
    void transformToAst() throws IOException {
        InputStream stream = this.getClass().getResourceAsStream("Shamroc.shamroc");
        ShamrocLoader subject = new ShamrocLoader(stream);
//        System.out.println(subject.getSyntaxErrors());
        assertTrue(subject.valid());
        Approvals.verify(subject.toAst());
    }
}
