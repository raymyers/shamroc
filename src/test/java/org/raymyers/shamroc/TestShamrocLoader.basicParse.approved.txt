(program (def (defLang lang Shamroc { (langItem (sumType Program : (sumTypeOption (nameOrType List < Def >))) ;) (langItem (sumType Def : (sumTypeOption (nameOrType DefLang))) ;) (langItem (record DefLang ( (recordField (nameOrType name) : (nameOrType String)) , (recordField (nameOrType List < LangItem >)) )) ;) (langItem (sumType LangItem : (sumTypeOption (nameOrType Record)) | (sumTypeOption (nameOrType SumType))) ;) (langItem (sumType Record : (sumTypeOption (nameOrType List < RecordField >))) ;) (langItem (record RecordField ( (recordField (nameOrType NameOrType)) , (recordField (nameOrType Optional < NameOrType >)) )) ;) (langItem (record NameOrType ( (recordField (nameOrType String)) , (recordField (nameOrType Optional < String >)) )) ;) (langItem (record SumType ( (recordField (nameOrType String)) , (recordField (nameOrType List < SumTypeOption >)) )) ;) (langItem (sumType SumTypeOption : (sumTypeOption (nameOrType Record)) | (sumTypeOption (nameOrType NameOrType))) ;) })) <EOF>)